src/pkg/fault/fault.go

```diff
@@ -0,0 +1,127 @@
+package fault
+
+import (
+	"sync"
+
+	"golang.org/x/exp/slices"
+)
+
+type errors struct {
+	mu *sync.Mutex
+
+	// err identifies non-recoverable errors.  This includes
+	// non-start cases (ex: cannot connect to client), hard-
+	// stop issues (ex: credentials expired) or conscious exit
+	// cases (ex: iteration error + failFast config).
+	err error
+
+	// errs is the accumulation of recoverable or iterated
+	// errors.  Eg: if a process is retrieving N items, and
+	// 1 of the items fails to be retrieved, but the rest of
+	// them succeed, we'd expect to see 1 error added to this
+	// slice.
+	errs []error
+
+	// if failFast is true, the first errs addition will
+	// get promoted to the err value.  This signifies a
+	// non-recoverable processing state, causing any running
+	// processes to exit.
+	failFast bool
```

Comment:
I'm a little hesitant about having this bool because it feels like we're mixing config/control flow info with error aggregation. Who is expected to set this? Are we passing these structs from BackupOp to GC/KW now because it also includes control flow information?

Reply:
> Who is expected to set this?

Whomever calls `fault.New(failFast)`, which should be the operation constructor.

> Are we passing these structs from BackupOp to GC/KW now because it also includes control flow information?

Yes, but not strictly because it includes control flow information.  We want the struct passed around to access its `Adder` interface any place we aggregate errors.  

Configuration dictates control flow, and it makes sense in this instance for control flow that depends on errors to be bound to said errors aggregation.